src/input/error.rs

```diff
@@ -18,6 +18,8 @@ pub enum AudioStreamError {
     Fail(Box<dyn Error + Send + Sync>),
     /// The operation was not supported, and will never succeed.
     Unsupported,
+    /// The Http request failed. If you get this you may be being rate limited
```

Comment:
This `Error` enum isn't really designed to be extended. A non-2xx request should map to an `AudioStreamError::Fail(...)` (I think a 503 with a Retry-after header is the exception, but that's out of scope). The reason we do this is so that the internals have a limited number of error cases to handle, many of which could be user code (source creation, or occur during decode).

Could you please make use of the `Fail` error variant rather than creating a new one?